File: 2021/April/Day 25/subtract-product-and-sum.py
# source : https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/submissions/
class Solution(object):

    def getDigitsList(self,n):

        digits = []
        while(n):
            digits.append(n%10);
            n = int(n/10)
        return digits

    def getProduct(self,l):
        f = 1;

        for i in l:
            f = f*i
        return f

    def subtractProductAndSum(self, n):
        """
        :type n: int
        :rtype: int
        """

        l = self.getDigitsList(n)
        p = self.getProduct(l)
        s = sum(l)
        return p-s

File: 2022/March/Day 21/subtractProductAndSum.cpp
// Source :https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/submissions/
class Solution {
public:
    int subtractProductAndSum(int n) {
        int s=0;
        int f=1;
        int d=0;
        while(n)
        {
            d=n%10;
            s+=d;
            f*=d;
            n=n/10;
        }
        return f-s;
    }
};
